util/db/repository_secrets.go

```diff
@@ -31,11 +31,19 @@ func (s *secretsRepositoryBackend) CreateRepository(ctx context.Context, reposit
 		},
 	}
 
-	s.repositoryToSecret(repository, repositorySecret)
+	repositoryToSecret(repository, repositorySecret)
 
 	_, err := s.db.createSecret(ctx, repositorySecret)
 	if err != nil {
 		if apierr.IsAlreadyExists(err) {
+			hasLabel, err := s.hasRepoTypeLabel(secName)
+			if err != nil {
+				return nil, status.Errorf(codes.Internal, err.Error())
+			}
+			if !hasLabel {
+				msg := "secret %q doesn't have the proper %q label: please fix the secret or delete it"
+				return nil, status.Errorf(codes.InvalidArgument, msg, secName, common.LabelKeySecretType)
```

Comment:
This is just a nitpick, but for the compiler (and/or linter) to catch issues with the format string, I think it would be wiser to either have `msg := fmt.Sprintf(...)` and use `status.Error()`, or just get rid of the intermediate variable `msg` and use `status.Errorf()` with the format string literal.

Reply:
The reason why I did that was to avoid having a too long line and avoid scrolling. Too bad the linter is not smart enough to verify the code with variables as well. I picked your first suggestion as it has better code readability. 👍🏻